TencentBlueKing/bk-user
# v2.3.4

### Bug Fixes

- 国际化 (#417)
- Custom profile sync add original `code` into `extras` (#420)
- Change CELERY_RESULT_BACKEND to django-db, instead of MQ or Redis (#419)
- Plugin init_settings add more detail in log info (#422)
- Add unique_together for user_settings_setting(category_id+meta_id) (#425)

### Miscellaneous Tasks

- 1.2.13

## [2.3.4-beta.20] - 2022-05-09

### Bug Fixes

- Token auth fail with different exception message (#408)
- Fix custom sync via http will raise IndexError if results is empty (#410)
- Logger remove sensitive info from data (#415)
- Custom sync from crontab use the same task_id (#411)
- 新增字段下拉显示问题 (#414)

### Miscellaneous Tasks

- V2.3.4-beta.20 (#416)

### Refactor

- Remove useless codes of esb (#409)

## [2.3.4-beta.19] - 2022-05-09

### Bug Fixes

- Migration setttings_meta create same key with the init_settings (#412)

## [2.3.4-beta.18] - 2022-05-06

### Bug Fixes

- Logo en image (#395)
- Fix http api as datasource wrong response check (#400)

### Styling

- Fix lints of isort (#396)

## [2.3.4-beta.17] - 2022-05-05

### Features

- Change multiple auth order, add one more bk-user api in login (#394)

## [2.3.4-beta.16] - 2022-04-22

### Bug Fixes

- Remove esb healthz check from api (#382)

## [2.3.4-beta.14] - 2022-04-21

### Miscellaneous Tasks

- Change chart version to 1.2.5 (#373)

## [2.3.4-beta.13] - 2022-04-19

### Bug Fixes

- 修改样式
- Fix http api datasource syncer (#368)
- 密码提交参数修改 (#370)

### Build

- V2.3.4-beta.13 (#372)

## [2.3.4-beta.12] - 2022-04-13

### Miscellaneous Tasks

- Update version to v2.3.4-beta.12

## [2.3.4-beta.11] - 2022-04-13

### Bug Fixes

- Password wrong 500 if use esb
- Call usermgr api response data None
- Root logger config wrong
- Unittest fail after change common logging settings
- Fix bug user_settings data is a list(should be dict)
- 手机号输入提示
- 国际化
- 修改参数类型

### Features

- Change version to v2.3.4-beta.10
- Enhance the logging while api 500
- Add request_id for saas call api
- Support both ee and raw apis
- Add bk_app_code from esb/apigateway in error log/audit log

### Miscellaneous Tasks

- Change version to 1.2.4

## [2.3.4-beta.10] - 2022-04-12

### Bug Fixes

- Add default api healthz check
- Change http check error body truncate

### Features

- Add shortcut get_client_with_app_code_app_secret_by_user
- Add /healthz/ for saas

## [2.3.4-beta.9] - 2022-04-08

### Bug Fixes

- Deploy fail on k8s

### Miscellaneous Tasks

- To v2.3.4-beta.9

## [2.3.4-beta.8] - 2022-04-07

### Bug Fixes

- Fix auth_login raise exception in some conditions

### Build

- Change version to 1.2.1

## [2.3.4-beta.7] - 2022-04-07

### Bug Fixes

- 添加搜索参数/密码规则校验
- Label selectors wrong
- 密码规则校验

### Miscellaneous Tasks

- Change Chart version

## [2.3.4-beta.6] - 2022-04-01

### Bug Fixes

- Fix missing import of REST_FRAMEWORK
- Fix app_code/app_secret header name wrong
- Add /healthz/ into auth url white list
- Fix esb get_public_key fail
- Add ENABLE_API_AUTH for api authentication in different versions
- Remove useless codes
- Fix comments
- 防御密码重置配置数据格式
- 修复二进制部署时 call_through_api 无法找到正确 Path 问题
- 修复二进制部署时 call_through_api 无法找到正确 Path 问题

### Features

- Call usermgr api via esb if usermgr api auth enabled
- Some bugs in helm deploy

## [2.3.4-beta.5] - 2022-03-28

### Bug Fixes

- 修复footer显示样式/国际化
- 修复目录创建后默认配置未创建的问题

### Api

- Add apigateway registry support

## [2.3.4-beta.4] - 2022-03-23

### Bug Fixes

- 豁免 SaaS 版本日志接口
- 豁免 SaaS 版本日志接口
- 修复 mypy 问题

## [2.3.4-beta.3] - 2022-03-23

### Bug Fixes

- 修复 no_page 时返回异常

## [2.3.4-beta.2] - 2022-03-23

### Bug Fixes

- 修复 swagger 解析错误
- 修复 swagger 解析错误
- 修复 no_page 时返回异常

### Miscellaneous Tasks

- 更新版本至 2.3.4-beta.2

### Revert

- 修复 swagger 解析错误

## [2.3.4-beta.1] - 2022-03-22

### Bug Fixes

- Add cache for dependencies
- Update cache path
- Add a condition for caching
- Add a condition for caching [WIP]
- Raise speed of unittest in actions [WIP]
- Raise speed of unittest in actions [WIP]
- Raise speed of unittest in actions [WIP]
- Raise speed of unittest in actions [WIP]
- 修复蓝鲸登录set_cookie中bk_token带引号问题 & 修正社区版 header & footer
- 修改登录框尺寸
- Adjust css of login plain form
- Fix lints
- Fix review comments
- BlueException missing message detail while logging or print
- 多字段查询优化 #73
- 密码设置增加配置 #238
- Check if SENTRY_DSN empty, instead of exists
- 搜索优化 73
- Django ugettext as _ conflict
- Footer显示版本号
- 修改登录页footer
- Title修改
- 为密码重置检测增加防御逻辑
- 暂时取消动态从 settings.yaml 中加载配置 & 修复部分文件 black 格式
- 补足login模块缺失依赖包
- 登录跳转路径添加尾/作为默认值，确保二进制部署时能够正确跳转
- Remove legacy notes

### Features

- 增加了通过 yaml 配置插件 SettingMeta 的能力
- Add sentry support for bk-user-api
- Add sentry support for bk-user-saas
- Add sentry support
- Add sentry_dsn in helm settings
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289
- Audit for password modification #289

### Miscellaneous Tasks

- Update chart version to v2.3.3-alpha.2
- Update version to 2.3.3
- 更新版本至 v2.3.4-beta.1

### Refactor

- Logger prints
- Unused data
- Remove unused codes
- Remove oauth2 codes, add some more logs
- Add some more loggers
- Remove static bkDialog-1.0 and layout_head.css

### Styling

- Use python3 super instead

### Enhance

- 为 v3 api 增加了查询长度限制

### Minor

- Change footer version
- Delete useless footer info

## [2.3.4-alpha.2] - 2022-03-02

### Bug Fixes

- Binary package can not upgrade due to some env lacking

## [2.3.4-alpha.1] - 2022-03-01

### Bug Fixes

- Login plugin not loaded normally
- Fix lints
- Fix review comments
- 修复当前开发配置文件模版问题
- BlueException missing message detail while logging or print
- Remove redundant logging while handle the exceptions
- Fix /ping always redirect to /ping/ in kube-probe
- 根据评审修改部分变量命名

### Features

- Support in query in v3 apis(only profiles)
- Add departments v3 apis
- Add v3 department apis in SaaS
- 密码过期邮件提醒功能 #117
- 支持密码连续字符等功能检测 #238

### Refactor

- Logger prints

### Styling

- Use python3 super instead

### Minor

- Change login page title
- Update saas dev tmpl setting
- 降低一些无用日志的等级
- Update default image registry to hub.bktencent.com
